src/VisualStudio/Core/Def/ColorSchemes/ColorSchemeApplier.Settings.cs

```diff
@@ -73,8 +85,10 @@ public void ApplyColorScheme(ColorSchemeName schemeName, ImmutableArray<Registry
             /// <summary>
             /// Get the color scheme that is applied to the configuration registry.
             /// </summary>
-            public ColorSchemeName GetAppliedColorScheme()
+            public async Task<ColorSchemeName> GetAppliedColorSchemeAsync(CancellationToken cancellationToken)
             {
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
```

Comment:
So we have this code here:

https://github.com/dotnet/roslyn/blob/1182298a7a78a2471e717f086d21b51b173fa2db/src/VisualStudio/Core/Def/Options/LocalUserRegistryOptionPersister.cs#L37-L42

Which implies we can get to local registry values without needing a UI thread switch. I'm not sure if VSRegistry.RegistryRoot() is just doing exactly what we do there or whether we can refactor that and reuse it?

Reply:
(this isn't wrong though, so better to keep the switch to UI thread if there is any uncertainty)